### PR TITLE
node: avoid clobbering terminal with binary data

### PR DESCRIPTION
**Pull request by leoluk**
**Stack**:
- #775
- #744
- #774 ⮜

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: node/cmd/guardiand/logging.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+	b := buf.Bytes()
+	for i := range b {
+		if unicode.IsControl(rune(b[i])) && !unicode.IsSpace(rune(b[i])) {
+			b[i] = '\x1A' // Substitute character
+		}
+	}
+
+	return buf, nil
+}
```
Comment (SEJeff) [inline node/cmd/guardiand/logging.go:28]: This exact same file with the exact same contents is already in #744. I'm sure git will dedup, but was this intentional?

File: node/cmd/guardiand/node.go [hunk 2]
```diff
@@ -245,12 +245,11 @@ func runNode(cmd *cobra.Command, args []string) {
...
-	cfg.Level = zap.NewAtomicLevelAt(zapcore.Level(lvl))
-	logger, err := cfg.Build()
-	if err != nil {
-		panic(err)
-	}
+	logger := zap.New(zapcore.NewCore(
+		consoleEncoder{zapcore.NewConsoleEncoder(
+			zap.NewDevelopmentEncoderConfig())},
+		zapcore.AddSync(zapcore.Lock(os.Stderr)),
+		zap.NewAtomicLevelAt(zapcore.Level(lvl))))
```
Comment (SEJeff) [inline node/cmd/guardiand/node.go:253]: This exact same diff is [here](https://github.com/certusone/wormhole/pull/744/files#diff-7f43e1a5bbb6083b5e470710e19e804d778f4a411b23fdbef1b2955f56b67bbdL248-R252) in #744. I think you messed up this spr stack

File: node/pkg/p2p/p2p.go [hunk 3]
```diff
@@ -293,7 +293,7 @@ func Run(obsvC chan *gossipv1.SignedObservation, sendC chan []byte, signedInC ch
 			err = proto.Unmarshal(envelope.Data, &msg)
 			if err != nil {
 				logger.Info("received invalid message",
-					zap.String("data", string(envelope.Data)),
+					zap.Binary("data", envelope.Data),
```
Comment (SEJeff) [inline node/pkg/p2p/p2p.go:296]: Same as above.

**Review by SEJeff — CHANGES_REQUESTED** [review]
It appears that the entire contents of this PR is already in #744, which I'd already approved. Git should merge it all in nicely, but I'm assuming this was not intentional?

**Comment by SEJeff** [comment]
I'd suggest closing this PR and just merging #774 since it has both this commit and the other commit inside it.

**Review by SEJeff — APPROVED** [review]

**Comment by leoluk** [comment]
✓ Commit merged in pull request [#775](https://github.com/certusone/wormhole/pull/775)

**Pull request closed without merging**